Add Condensed Contact Form in Footer

Cameron-Porter opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
Full update...Updated to only use contact api and not add a second redundant one for the footer contact. Removed the requirement for name and phone, but placed default values for when the footer contact is used. HTML semantics should still cover the contact form as a whole.

## Related Issue
#259 

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->

## How Has This Been Tested?
Updated tests and all passed for footer contact form. The contact form did have an issue and I was able to update it to check only a param required by both forms. Passing now. But please let me know if we need to revisit this.

## Screenshots (if appropriate):

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

ajames14-chwy commented on src/pages/api/contactFooter.js:
```diff
@@ -0,0 +1,34 @@
+import axios from 'axios'
```
@Cameron-Porter  is there a way we could reuse the existing contact form api vs creating a new route for this? I understand there are different requirements for each form but seems like something we should be able manage in one route. Thoughts?

Cameron-Porter replied:
I could try and implement the additions within the api/contact.js file if that is what you mean. Otherwise, I could create a new test file for the separate api file.

andrew-t-james replied:
Yeah I think reuse of the current api rout and updating existing tests to cover the changes would acceptable.

Cameron-Porter replied:
Ok, I will give that a shot and send it back to see if it works. Great learning opp!

AaronAstle replied:
@Cameron-Porter I think I am seeing what @andrew-t-james, lets look in to refactoring `contact.js` in lieu of adding this additional file.  We want to aim for DRY ( Dont Repeat Yourself ) on the code.  So much of this duplicates efforts in the `contact.js` file.

ajames14-chwy commented on src/pages/api/contactFooter.js:
```diff
@@ -0,0 +1,34 @@
+    })
+  }
+
+  const text = [`Name: \`${name}\``, `\nMessage: \n\`\`\`${message}\`\`\``].join()
```
If the above is not doable this would need test coverage as well.

Cameron-Porter commented:
Pushed some changes. Updated so footer contact form uses same API endpoint as the normal contact form. Updated the main description with additional details. Please let me know if I need to revisit or refactor anything.

@andrew-t-james / @ajames14-chwy

andrew-t-james commented on tests/components/Forms/ContactFormFooter.test.js:
```diff
@@ -0,0 +1,164 @@
+import { render, fireEvent, waitFor } from '@testing-library/react'
+import { ContactFormFooter } from '@/components/Forms'
```
Please remove this file as well they are not needed.

andrew-t-james commented on src/components/Forms/ContactFormFooter/ContactFormFooter.js:
```diff
@@ -0,0 +1,96 @@
+  const [loading, setLoading] = useState(false)
+  const { register, handleSubmit, errors, reset } = useForm()
+
+  const onSubmit = async (formData, e) => {
```
Please remove this file completely. Since we are using the contact form this should be no longer needed.

Cameron-Porter commented on src/components/Forms/ContactFormFooter/ContactFormFooter.js:
```diff
@@ -0,0 +1,97 @@
+import { useForm } from 'react-hook-form'
+import { FormAlert, onSubmitSuccess, onSubmitError } from '../'
+
+function ContactFormFooter() {
```
Hey @andrew-t-james, I will take a look and see what I can come up with.

Cameron-Porter commented:
Closing due to multiple files populating from merge. Re-pushing new PR for Andrew's review.
